src/zcl_text2tab_parser.clas.testclasses.abap

```diff
@@ -374,36 +385,69 @@ class ltcl_text2tab_parser_test implementation.
     split dummy_head at c_tab into table lt_header_exp.
 
     try.
-      o->parse(
-        exporting
-          i_data        = l_string
-          i_strict      = abap_false
-        importing
-          e_container   = dummy_tab_act
-          e_head_fields = lt_header_act ).
-      cl_abap_unit_assert=>assert_equals( act = dummy_tab_act exp = dummy_tab_exp ).
-      cl_abap_unit_assert=>assert_equals( act = lt_header_act exp = lt_header_exp ).
-    catch zcx_text2tab_error into lx.
-      cl_abap_unit_assert=>fail( lx->get_text( ) ).
+        o->parse(
+          exporting
+            i_data        = l_string
+            i_strict      = abap_false
+          importing
+            e_container   = dummy_tab_act
+            e_head_fields = lt_header_act ).
+        cl_abap_unit_assert=>assert_equals( act = dummy_tab_act exp = dummy_tab_exp ).
+        cl_abap_unit_assert=>assert_equals( act = lt_header_act exp = lt_header_exp ).
+      catch zcx_text2tab_error into lx.
+        cl_abap_unit_assert=>fail( lx->get_text( ) ).
     endtry.
 
   endmethod.  "parse
 
+  method parse_time.
+    data: l_exp_result        type ty_dummy_with_time,
+          l_act_result        type ty_dummy_with_time,
+          l_with_valid_time   type string,
+          l_with_invalid_time type string,
+          cut                 type ref to zcl_text2tab_parser.
+
+    get_dummy_data_with_time( importing
+      e_exp_result = l_exp_result
+      e_with_valid_time = l_with_valid_time
+      e_with_invalid_time = l_with_invalid_time
+    ).
+
+    cut = zcl_text2tab_parser=>create( l_exp_result ).
+
+    " not successfull parsing
+    try.
+        cut->parse( exporting i_data = l_with_invalid_time
+          importing e_container = l_act_result ).
+        cl_abap_unit_assert=>fail( msg = 'no exception when given invalid time' ).
+      ##no_handler
+      catch zcx_text2tab_error.
```

Comment:
please add the check of the exception code (that it was exactly the expected exception)